src/content/en/2019/javascript.md

```diff
@@ -0,0 +1,338 @@
+scripts with the attribute normally. Using this approach can allow developers to
+[send modern code to modern browsers for faster page loads](https://web.dev/serve-modern-code-to-modern-browsers/).
+
+So, **how many sites use `nomodule` for scripts on their page?**
+
+<insert graphic of metric 01_13>
+
+Similarly, very few sites (**~0.5% - 0.8%**) use the `nomodule` attribute for any scripts.
+
+# Preload and prefetch
+
+**[Preload**](https://developer.mozilla.org/en-US/docs/Web/HTML/Preloading_content) and
+**[prefetch**](https://developer.mozilla.org/en-US/docs/Web/HTTP/Link_prefetching_FAQ) are
+directives which enable you to aid the browser in determining what resources need to be
+downloaded.
+
+-  Preloading a resource with `<link rel="preload"> `tells the browser to download this resource
+   as soon as possible. This is especially helpful for critical resources which are discovered late
+   in the page loading process (e.g., javascript located at the bottom of your HTML) and are
+   otherwise downloaded last.
+-  Using `<link rel="prefetch">`  tells the browser to take advantage of any idle time it has to
+   fetch these resources needed for future navigations
+
+**So, how many sites use preload and prefetch directives?**
+
+<insert graphic of metric 01_14>
+
+For all sites measured in HTTP Archive, **14.33%** of desktop sites and **14.84%** of mobile sites
+use `<link rel="preload"> `for scripts on their page.
+
+For prefetch:
+
+<insert graphic of metric 01_16>
+
+For both mobile and desktop, **0.08% (~0.1%)** of pages leverage prefetch for any of their
+scripts.
+
+# Newer APIs
+
+JavaScript continues to evolve as a language. A new version of the language standard itself, known
+as ECMAScript, is released every year with new APIs and features passing proposal stages to become a
+part of the language itself.
+
+With HTTP Archive, we can take a look at any newer API that is supported (or is about to be) and see
+how widespread its usage is. These APIs may already be used in browsers that support them _or_ with
+an accompanying polyfill to make sure they still work for every user.
+
+**How many sites use the following APIs?**
+
+-  [Atomics](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Atomics)
+-  [Intl](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl)
+-  [Proxy](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy)
+-  [SharedArrayBuffer](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/SharedArrayBuffer)
+-  [WeakMap](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/WeakMap)
+-  [WeakSet](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/WeakSet)
+
+<insert graphic of metric 01_18>
+
+Atomics (**0.38%**) and SharedArrayBuffer (**0.2%**) are barely visible on this chart since they are
```

Comment:
Should we include `Worker` usage in the chart? It would be interesting to compare it with these two features, as workers are pretty much the only usecase scenario for them.

Reply:
Same probably goes for `WebAssembly` API, as I don't think people use SAB and Atomics much from pure JavaScript.

Reply:
I just feel like mentioning SAB and Atomics % metrics without context (the "high bar" of all Workers) is a bit pointless, as these numbers don't say much about amount of code where they could be potentially used in the first place.